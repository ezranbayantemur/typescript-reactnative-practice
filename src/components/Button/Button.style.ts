import {StyleSheet} from 'react-native';
import Colors from '../../styles/colors';
import Spacing from '../../styles/spacing';

export default {
  primary: StyleSheet.create({
    container: {
      backgroundColor: Colors.purple,
      padding: Spacing.normal,
      margin: Spacing.normal,
      borderRadius: 5,
      justifyContent: 'center',
      alignItems: 'center',
    },
    title: {
      color: Colors.white,
      fontSize: 15,
    },
  }),
  outline: StyleSheet.create({
    container: {
      borderWidth: 1,
      borderColor: Colors.purple,
      padding: Spacing.normal,
      margin: Spacing.normal,
      borderRadius: 5,
      justifyContent: 'center',
      alignItems: 'center',
    },
    title: {
      color: Colors.purple,
      fontSize: 15,
    },
  }),
};
