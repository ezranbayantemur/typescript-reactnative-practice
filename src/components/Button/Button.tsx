import React, {FC} from 'react';
import {
  Text,
  TouchableOpacity,
  TouchableOpacityProps,
  ActivityIndicator,
} from 'react-native';

interface ButtonProps extends TouchableOpacityProps {
  title: string;
  loading?: boolean;
}

const Button: FC<ButtonProps> = ({title, loading, ...otherProps}) => {
  return (
    <TouchableOpacity
      testID="button-touchable"
      disabled={loading}
      {...otherProps}>
      {loading ? (
        <ActivityIndicator size="large" testID="button-indicator" />
      ) : (
        <Text testID="button-text">{title}</Text>
      )}
    </TouchableOpacity>
  );
};

export default Button;
