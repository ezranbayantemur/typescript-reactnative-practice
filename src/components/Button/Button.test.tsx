import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Button from '.';

test('should render correctly', () => {
  const comp = render(<Button title="" />);
  expect(comp).toMatchSnapshot();
});

test('should render given title', () => {
  const testTitle = 'test';
  const comp = render(<Button title={testTitle} />);
  const buttonTitle = comp.getByTestId('button-text').children[0];
  expect(buttonTitle).toBe(testTitle);
});

test('should trigger press event', () => {
  const mockFunction = jest.fn();
  const comp = render(<Button title="" onPress={mockFunction} />);
  const buttonTouchable = comp.getByTestId('button-touchable');
  fireEvent(buttonTouchable, 'press');
  expect(mockFunction).toBeCalled();
});

test('should render indicator on loading state', () => {
  const comp = render(<Button title="" loading />);
  const buttonIndicator = comp.getByTestId('button-indicator');
  expect(buttonIndicator).toBeTruthy();
});

test('should not trigger press event on loading state', () => {
  const mockFunction = jest.fn();
  const comp = render(<Button title="" loading onPress={mockFunction} />);
  const buttonTouchable = comp.getByTestId('button-touchable');
  fireEvent(buttonTouchable, 'press');
  expect(mockFunction).not.toBeCalled();
});
