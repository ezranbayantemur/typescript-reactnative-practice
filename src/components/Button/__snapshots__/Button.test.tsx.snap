// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#8e24aa",
      "borderRadius": 5,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="button-touchable"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 15,
      }
    }
    testID="button-text"
  >
    
  </Text>
</View>
`;
